modules/ServerAPI/src/server/routes/AuthorRoutes.coffee

```diff
@@ -964,4 +966,78 @@ exports.signupNewAuthorInternal = (author, cb) ->
 			cb json, null
 		else
 			cb null, json
+	)
+
+exports.updateAuthorRoles = (req, resp) ->
+	if req.session?.passport?.user?
+		user = req.session.passport.user
+	else
+		user =
+			username: 'anonymous'
+			roles: []
+	exports.updateAuthorRolesInternal req.body.newAuthorRoles, req.body.authorRolesToDelete, user, (err, response) ->
+		if err?
+			resp.statusCode = 500
+			resp.json err
+		else
+			resp.json response
+
+exports.updateAuthorRolesInternal = (newAuthorRoles, authorRolesToDelete, user, callback) ->
```

Comment:
`updateAuthorRolesInternal` is pretty similar to sync roles except it doesn't do this check.  Can we consolidate the code a little?  If not I think it would be good to add a note hear on the difference.

Reply:
Thanks, this led to a nice refactor.